### DISCUSSION TITLE
How to Check the Version Number

### DISCUSSION THREAD

**Discussion by shawn8888** [haugene/docker-transmission-openvpn#2630] [Q&A] [answered]
I would like to know the version of docker-transmission-openvpn and the corresponding version of Transmission it is utilizing. I believe the latest version for the docker container is v5.0.2, and for Transmission, it is v4.0.3. Could you please guide me on how to confirm these versions?

Thank you!

**Comment by pkishino** [comment]
For this container: 
https://github.com/haugene/docker-transmission-openvpn/releases
Check what image you are using in your compose or run command..
Regarding transmission, depends on your UI, but I believe it’s also in the logs or home directory

**Reply by shawn8888** [reply]
what command should I run to confirm the version I am running?
I use Transmission Remote GUI to connect. It doesn't show what version of Transmission it connects to.

![image](https://github.com/haugene/docker-transmission-openvpn/assets/8271101/d01f9486-0211-44ae-9dfa-42b8ab040f77)

**Reply by pkishino** [reply]
if you use the official UI it shows up on the web..otherwise in the logs..
But if you use a specific :<tag> of this container, the release notes specify the version..

**Reply by shawn8888** [reply]
I do download the "latest" tag image. 
"transmission-remote -V" works for checking the Transmission version. It would be nice to have a way to verify the version number of the container as well.

**Comment by edgd1er** [comment, marked as answer]
from within the container:
```
root@f40cde7227f4:/app# transmission-remote -V
transmission-remote 4.0.3 (6b0e49bbb2)
root@f40cde7227f4:/app# transmission-daemon -V
transmission-daemon 4.0.3 (6b0e49bbb2)
```

or with another web ui:
![image](https://github.com/haugene/docker-transmission-openvpn/assets/3601462/9b3b8c59-4442-4922-9829-46f9e90af62a)

**Reply by shawn8888** [reply]
Thank you! It works!
Any idea how to verify the docker container's version number?

**Comment by pkishino** [comment]
Revision is in the container logs..
Instead of grabbing :latest it is recommended to use a tagged release..

**Comment by shawn8888** [comment]
Sorry pkishino, but I cannot find it in the container logs.
I used the command 
docker logs transmission1 -f --tail 100
I don't see anything related to the version number.

**Comment by pkishino** [comment]
It’s in the logs when the container starts up

**Reply by shawn8888** [reply]
I restarted the container and didn't find it.
What's yours look like? Just the line with the version#.

**Reply by pkishino** [reply]
yes it is there..its the first line printed:
Starting container with revision: 1103172c3288b7de681e2fb7f1378314f17f66cf

**Reply by shawn8888** [reply]
How to locate the first line of the log after a restart?
I also tried to search key word "revision" and I found nothing:
`docker logs transmission1 -f --tail 100 2>&1  |grep revision`
What key word should I look for?

Also, why do you not recommend :latest tag? I thought it means the latest stable version. Is that not true?
I just checked 
https://registry.hub.docker.com/r/haugene/transmission-openvpn/tags
And both tag latest and 5.0.2 show "last pushed 2 months ago". It looks like they are the same thing.

**Comment by pkishino** [comment]
Remove the tail 100…
Stop container, Clear the logs, then start it again.. is the First entry

**Comment by shawn8888** [comment]
@pkishino Thank you!
I finally see it in logs! :
`+ echo 'Starting container with revision: a4d65774f855a04070766e53346661f48c76fa0e'`
However, it looks like a hash code rather than a plain version number, such as 5.0.2.

**Comment by pkishino** [comment]
It’s the git commit hash..
Again, as I mentioned before. Use :<tag> instead of :latest instead. Then
you get the tagged version.
